test/rbac_test.js

```diff
@@ -0,0 +1,345 @@
+        ),
+        alice,
+    );
+
+    // Add role[2] --> perm[3]
+    await contractTransaction(
+        contract.tx.addRoleToPermission(
+            { value: 0, gasLimit: 100000 * 1000000 }, ROLE_ADDRS[2], PERM_ADDRS[3],
+        ),
+        alice,
+    );
+
+    console.log('--- Test readPermissions');
+    // Permission exist
+    // Permission ROLE_ADDRS[0] has PERM_ADDRS[0]
+    callValue = await contract.query.readPermissions(
+        alice.address,
+        { value: 0, gasLimit: -1 },
+        ROLE_ADDRS[0],
+    );
+    assert.equal(callValue.output.length, 2);
+    callValue.output.forEach((element) => {
+        assert.notEqual([PERM_ADDRS[0], PERM_ADDRS[1]].indexOf(element.toHex()), -1);
+    });
+
+    callValue = await contract.query.readPermissions(
+        alice.address,
+        { value: 0, gasLimit: -1 },
+        ROLE_ADDRS[1],
+    );
+    assert.equal(callValue.output.length, 1);
+    assert.equal(callValue.output[0].toHex(), PERM_ADDRS[2]);
+
+    callValue = await contract.query.readPermissions(
+        alice.address,
+        { value: 0, gasLimit: -1 },
+        ROLE_ADDRS[2],
+    );
+    assert.equal(callValue.output.length, 1);
+    assert.equal(callValue.output[0].toHex(), PERM_ADDRS[3]);
+
+    // Permission ROLE_ADDRS[3] has no permission
+    callValue = await contract.query.readPermissions(
+        alice.address,
+        { value: 0, gasLimit: -1 },
+        ROLE_ADDRS[3],
+    );
+    assert.equal(callValue.output.length, 0);
+
+    console.log('--- Test checkAccess');
+    // Permission exist
+    // Permission ROLE_ADDRS[0] has PERM_ADDRS[0]
+    callValue = await contract.query.checkAccess(
+        alice.address,
+        { value: 0, gasLimit: -1 },
+        USER_ADDRS[0],
+        PERM_ADDRS[0],
+    );
+    assert.equal(callValue.output.valueOf(), true);
+
+    callValue = await contract.query.checkAccess(
+        alice.address,
+        { value: 0, gasLimit: -1 },
+        USER_ADDRS[0],
+        PERM_ADDRS[1],
+    );
+    assert.equal(callValue.output.valueOf(), true);
+
+    callValue = await contract.query.checkAccess(
+        alice.address,
+        { value: 0, gasLimit: -1 },
+        USER_ADDRS[0],
+        PERM_ADDRS[2],
+    );
+    assert.equal(callValue.output.valueOf(), true);
+
+    callValue = await contract.query.checkAccess(
+        alice.address,
+        { value: 0, gasLimit: -1 },
+        USER_ADDRS[0],
+        PERM_ADDRS[3],
+    );
+    assert.equal(callValue.output.valueOf(), true);
+
+    callValue = await contract.query.checkAccess(
+        alice.address,
+        { value: 0, gasLimit: -1 },
+        USER_ADDRS[0],
+        PERM_ADDRS[4],
+    );
+    assert.equal(callValue.output.valueOf(), false);
+
+    callValue = await contract.query.checkAccess(
+        alice.address,
+        { value: 0, gasLimit: -1 },
+        USER_ADDRS[1],
+        PERM_ADDRS[0],
+    );
+    assert.equal(callValue.output.valueOf(), false);
```

Comment:
the api connection should be closed here else it will be left hanging and affect the node server
`api.disconnect();`